controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -843,23 +840,21 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(const std::string &sr
             LOG(INFO) << "Skipping " << bss_info_conf.ssid << " due to operclass mismatch";
             continue;
         }
-        if (!(tlvwscM1->authentication_type_flags_attr().data &
-              uint16_t(bss_info_conf.authentication_type))) {
+        if (!(m1->auth_type_flags() & uint16_t(bss_info_conf.authentication_type))) {
             LOG(INFO) << std::hex << "Auth mismatch for " << bss_info_conf.ssid << ": get 0x"
-                      << tlvwscM1->authentication_type_flags_attr().data << " need 0x"
+                      << m1->auth_type_flags() << " need 0x"
                       << uint16_t(bss_info_conf.authentication_type);
         }
-        if (!(tlvwscM1->encryption_type_flags_attr().data &
-              uint16_t(bss_info_conf.encryption_type))) {
+        if (!(m1->encr_type_flags() & uint16_t(bss_info_conf.encryption_type))) {
             LOG(INFO) << std::hex << "Encr mismatch for " << bss_info_conf.ssid << ": get 0x"
-                      << tlvwscM1->encryption_type_flags_attr().data << " need 0x"
+                      << m1->encr_type_flags() << " need 0x"
                       << uint16_t(bss_info_conf.encryption_type);
         }
         if (num_bsss >= radio_basic_caps->maximum_number_of_bsss_supported()) {
             LOG(INFO) << "Configured #BSS exceeds maximum for " << al_mac << " radio " << ruid;
             break;
         }
-        if (!autoconfig_wsc_add_m2(tlvwscM1, &bss_info_conf)) {
+        if (!autoconfig_wsc_add_m2(*m1, &bss_info_conf)) {
```

Comment:
Why not change the argument of this method to reference as well?

Reply:
m1 is created in this method, it is not passed as an argument.